tools/ea/Ea/Genesis.hs

```diff
@@ -49,87 +53,51 @@ module Ea.Genesis
 import Control.Lens
 
 import Data.Text
+import Data.Word
 
 import Chainweb.Graph
 import Chainweb.Test.TestVersions
 import Chainweb.Version
 import Chainweb.Version.Development
+import Chainweb.Version.FastDevelopment
 import Chainweb.Version.Mainnet
 import Chainweb.Version.Testnet
 
 
 -- ---------------------------------------------------------------------- --
 -- Genesis Tx Data
 
--- | Type-safe representation of chain ids
--- for a 10-chain graph.
---
-data GChainId
-    = Zero
-    | One
-    | Two
-    | Three
-    | Four
-    | Five
-    | Six
-    | Seven
-    | Eight
-    | Nine
-    | N
-    | KAD
-    deriving (Eq, Ord, Enum)
-
-instance Show GChainId where
-  show = \case
-    Zero -> "0"
-    One -> "1"
-    Two -> "2"
-    Three -> "3"
-    Four -> "4"
-    Five -> "5"
-    Six -> "6"
-    Seven -> "7"
-    Eight -> "8"
-    Nine -> "9"
-    N -> "N"
-    KAD -> "KAD"
+-- A range of chain IDs [l,r]
+data ChainIdRange
+    = ChainIdRange !Word32 !Word32
+    deriving (Eq, Ord)
+
+chainIdRangeTag :: ChainIdRange -> String
+chainIdRangeTag (ChainIdRange l u)
+    | l == u = show l
+    | otherwise = show l <> "to" <> show u
 
 -- | Genesis transaction record
 --
 data Genesis = Genesis
-  { _version :: ChainwebVersion
-    -- ^ chainweb version (e.g. Testnet04)
-  , _tag :: Text
-    -- ^ Module name tag
-  , _txChainId :: GChainId
-    -- ^ chain id
-  , _coinbase :: Maybe FilePath
-    -- ^ filepath to coinbase yaml
-  , _keysets :: Maybe FilePath
-    -- ^ filepath to keyset yaml
-  , _allocations :: Maybe FilePath
-    -- ^ filepath to allocation yaml
-  , _namespaces :: Maybe FilePath
-    -- ^ filepath to namespace yaml
-  } deriving (Eq, Ord, Show)
-
--- | A 'Lens'' into the transaction chain id
--- of a genesis payload
---
-txChainId :: Lens' Genesis GChainId
-txChainId = lens _txChainId (\t b -> t { _txChainId = b })
-
--- | A 'Lens'' into the transaction allocation filepath
--- of a genesis payload
---
-allocations :: Lens' Genesis (Maybe FilePath)
-allocations = lens _allocations (\t b -> t { _allocations = b })
-
--- | A 'Lens'' into the transaction allocation filepath
--- of a genesis payload
---
-coinbase :: Lens' Genesis (Maybe FilePath)
-coinbase = lens _coinbase (\t b -> t { _coinbase = b })
+    { _version :: ChainwebVersion
+      -- ^ chainweb version (e.g. Testnet04)
+    , _tag :: Text
+      -- ^ Module name tag
+    , _txChainIds :: ChainIdRange
+      -- ^ chain id
+    , _coinbase :: Maybe FilePath
+      -- ^ filepath to coinbase yaml
+    , _keysets :: Maybe FilePath
+      -- ^ filepath to keyset yaml
+    , _allocations :: Maybe FilePath
+      -- ^ filepath to allocation yaml
+    , _namespaces :: Maybe FilePath
+      -- ^ filepath to namespace yaml
+    , _coinContract :: [FilePath]
+    } deriving (Eq, Ord) -- Show)
+
+makeLensesFor [("_" <> fn, fn) | fn <- ["txChainIds", "coinbase", "keysets", "allocations", "namespaces", "coinContract"]] ''Genesis
```

Comment:
Nice. This could one day be a helper that leverages e.g. [reifyRecord](https://hackage.haskell.org/package/th-abstraction-0.5.0.0/docs/Language-Haskell-TH-Datatype.html#v:reifyRecord)